optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/tri/BavetFilterTriTuple.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bavet.tri;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class BavetFilterTriTuple<A, B, C> extends BavetAbstractTriTuple<A, B, C> {
+
+    private final BavetFilterTriNode<A, B, C> node;
+    private final BavetAbstractTriTuple<A, B, C> parentTuple;
+
+    protected List<BavetAbstractTriTuple<A, B, C>> childTupleList = null;
+
+    public BavetFilterTriTuple(BavetFilterTriNode<A, B, C> node, BavetAbstractTriTuple<A, B, C> parentTuple) {
+        this.node = node;
+        this.parentTuple = parentTuple;
+        childTupleList = new ArrayList<>();
```

Comment:
Any reason why this is not initialized directly in-line on line 27?

Reply:
The goal is to get a good starting capacity for these array lists, for performance reasons,
by using the childNodeList.size() of the node, multiplied by a good constant.
But there's another perf improvement that will heavily impact this.
Anyway, for now I added in createTuple() methods:
        // TODO Use childNodeList.size() to the tuple's childTupleList's capacity